include/boost/random/faure.hpp

```diff
@@ -0,0 +1,369 @@
+/* boost random/faure.hpp header file
+ *
+ * Copyright Justinas Vygintas Daugmaudis 2010-2017
+ * Distributed under the Boost Software License, Version 1.0. (See
+ * accompanying file LICENSE_1_0.txt or copy at
+ * http://www.boost.org/LICENSE_1_0.txt)
+ */
+
+#ifndef BOOST_RANDOM_FAURE_HPP
+#define BOOST_RANDOM_FAURE_HPP
+
+#include <boost/random/detail/qrng_base.hpp>
+
+#include <vector>
+#include <algorithm>
+
+#include <cmath>
+
+#include <boost/assert.hpp>
+
+#include <sstream>
+
+//!\file
+//!Describes the quasi-random number generator class template faure.
+
+namespace boost {
+namespace random {
+
+/** @cond */
+namespace detail {
+namespace fr {
+
+// There is no particular reason why 187 first primes were chosen
```

Comment:
There is a prime table given in `boost/math/special_functions/prime.hpp`. Presumably this should be used?

Reply:
We may also drop a sieve of Eratosthenes into boost.integer. Could this get us to arbitrary dimension?

Reply:
Oh, I will gladly use some function from math header! But note that I need to know a lower bound prime for a given number `n`, as opposed to just getting an `n`-th prime. If something like this could be implemented in math section, it would be nice (besides, `prime` and `lower_bound_prime` sounds like very complementary functions).

As for an arbitrary dimension, it's a bit more complicated than just plugging in a sufficiently big prime. van der Corput sequence computation would also need to be at infinite precision. It is possible, but costly.